### PR TITLE
Add ability to get groups by name with fields option (#786)

### PR DESCRIPTION
**Pull request by wadahiro**
This PR resolves issue #786.

### CODE REVIEW

**Comment by wadahiro** [comment]
@sujaygarlanka Could you review this pull request?

File: src/main/java/com/box/sdk/BoxGroup.java [hunk 1]
```diff
@@ -163,6 +163,34 @@ public BoxGroup(BoxAPIConnection api, String id) {
...
+     * @return     an iterable containing info about all the groups.
+     */
+    public static Iterable<BoxGroup.Info> getAllGroupsByName(final BoxAPIConnection api, String name,
+                                                             String... fields) {
+        final QueryStringBuilder builder = new QueryStringBuilder();
+        if (name == null || name.trim().isEmpty()) {
+            throw new BoxAPIException("Searching groups by name requires a non NULL or non empty name");
+        } else {
+            builder.appendParam("name", name);
+            if (fields.length > 0) {
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxGroup.java:181]:
```suggestion
            if (fields != null && fields.length > 0) {
```

File: src/main/java/com/box/sdk/BoxGroup.java [hunk 2]
```diff
@@ -163,6 +163,34 @@ public BoxGroup(BoxAPIConnection api, String id) {
...
 
+    /**
+     * Gets an iterable of all the groups in the enterprise that are starting with the given name string.
+     * @param  api the API connection to be used when retrieving the groups.
+     * @param  name the name prefix of the groups. If the groups need to searched by full name that has spaces,
+     *              then the parameter string should have been wrapped with "".
+     * @param fields the fields to retrieve.
+     * @return     an iterable containing info about all the groups.
+     */
+    public static Iterable<BoxGroup.Info> getAllGroupsByName(final BoxAPIConnection api, String name,
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxGroup.java:174]: Can you please have the method above this one call this one?

```
public static Iterable<BoxGroup.Info> getAllGroupsByName(final BoxAPIConnection api, String name) {
     return getAllGroupsByName(api, name, null);
 }
```

**Comment by PJSimon** [comment]
Hi @wadahiro, can you please update your forked branch with the latest changes on master?  Thanks!

**Comment by wadahiro** [comment]
@PJSimon Thanks you for the review.
I fixed the code you pointed out and rebased it on master. Could you check again?

File: src/main/java/com/box/sdk/BoxGroup.java (hunk 1 above)
Comment (PJSimon) [reply src/main/java/com/box/sdk/BoxGroup.java]: @wadahiro This looks good, thanks!

File: src/main/java/com/box/sdk/BoxGroup.java (hunk 2 above)
Comment (PJSimon) [reply src/main/java/com/box/sdk/BoxGroup.java:162]: @wadahiro Can you make this change please?  Then I'll be able to approve.  Thanks!

Comment (PJSimon) [reply src/main/java/com/box/sdk/BoxGroup.java:162]: @wadahiro Please disregard - my system wasn't refreshing or something.  I see the change now.  Thanks!

**Pull request merged**